x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx

```diff
@@ -0,0 +1,331 @@
+    | undefined
+  >();
+  const params = slo.indicator.params as KQLCustomIndicator['params'];
+  const groupBy = slo.groupBy;
+  const groupings = slo.groupings;
+  const { index } = params;
+  const { data: dataViews = [] } = useFetchDataViews();
+
+  useEffect(() => {
+    const getDataView = async () => {
+      const getDataViewByIndexPattern = (indexPattern: string) =>
+        dataViews.find((dataView0) => dataView0.title === indexPattern);
+
+      const dataViewId = getDataViewByIndexPattern(index)?.id;
+      if (dataViewId) {
+        const sloDataView = await dataViewsService.get(dataViewId);
+        setDataView(sloDataView);
+        getQuery();
+      }
+    };
+
+    const getQuery = () => {
+      const esSearchRequest = getESQueryForLogRateAnalysis(
+        params,
+        groupBy,
+        groupings
+      ) as QueryDslQueryContainer;
+      if (esSearchRequest) {
+        setEsSearchQuery(esSearchRequest);
+      }
+    };
+    getDataView();
+  }, [index, dataViews, params, dataViewsService, groupBy, groupings]);
+
+  // Identify `intervalFactor` to adjust time ranges based on alert settings.
+  // The default time ranges for `initialAnalysisStart` are suitable for a `1m` lookback.
+  // If an alert would have a `5m` lookback, this would result in a factor of `5`.
+  // If an alert is just starting, the visible time range to look back might not cover the
+  // long window yet, only then we'll take the short window to look back,
+  // otherwise it will be the long window.
+  const alertActionGroup =
+    alert.fields['kibana.alert.action_group'] !== 'recovered'
+      ? alert.fields['kibana.alert.action_group']
+      : LOW_PRIORITY_ACTION_ID;
+  const sloWindows = alert.fields['kibana.alert.rule.parameters']!.windows as WindowSchema[];
+  const relatedWindow = sloWindows.find((window) => window.actionGroup === alertActionGroup);
+
+  const longWindowValue = relatedWindow?.longWindow.value;
+  const longWindowUnit = relatedWindow?.longWindow.unit;
+  const longWindowLookbackDuration =
+    longWindowValue && longWindowUnit
+      ? moment.duration(longWindowValue as number, longWindowUnit as any)
+      : moment.duration(1, 'm');
+  const longWindowLookbackDurationAsSeconds = longWindowLookbackDuration.asSeconds();
+
+  const shortWindowValue = relatedWindow?.shortWindow.value;
+  const shortWindowUnit = relatedWindow?.shortWindow.unit;
+  const shortWindowLookbackDuration =
+    shortWindowValue && shortWindowUnit
+      ? moment.duration(shortWindowValue as number, shortWindowUnit as any)
+      : moment.duration(1, 'm');
+  const shortWindowLookbackDurationAsSeconds = shortWindowLookbackDuration.asSeconds();
+
+  const actionGroup = getActionGroupFromReason(alert.reason);
+  const actionGroupWindow = (
+    (alert.fields[ALERT_RULE_PARAMETERS]?.windows ?? []) as WindowSchema[]
+  ).find((window: WindowSchema) => window.actionGroup === actionGroup);
```

Comment:
We already have `alertActionGroup` and `relatedWindow` from above. Why do we need to find it again based on the reason?

Reply:
Here's relevant [discussion](https://github.com/mgiota/kibana/pull/1/files#r1567170831)

Reply:
This is already fixed [here](https://github.com/mgiota/kibana/pull/2)